gensim/models/doc2vec.py

```diff
@@ -572,22 +570,18 @@ def __init__(self, documents=None, input_streams=None, dm_mean=None, dm=1, dbow_
         self.docvecs = docvecs or Doc2VecKeyedVectors(self.vector_size, docvecs_mapfile)
 
         self.comment = comment
-        if documents is not None or input_streams is not None:
-            self._check_input_data_sanity(data_iterable=documents, data_iterables=input_streams)
-            if input_streams is not None:
-                if not isinstance(input_streams, (tuple, list)):
-                    raise TypeError("You must pass tuple or list as the input_streams argument.")
-                if any(isinstance(stream, GeneratorType) for stream in input_streams):
-                    raise TypeError("You can't pass a generator as any of input streams. Try an iterator.")
-                if any(isinstance(stream, TaggedLineDocument) for stream in input_streams):
-                    warnings.warn("Using TaggedLineDocument in multistream mode can lead to incorrect results "
-                                  "because of tags collision.")
+
+        assert corpus_file is None, "corpus_file for Doc2Vec is not supported yet."
```

Comment:
should be RuntimeError (instead of exception)

Reply:
done